### PR TITLE
Fix Documentation for data structures

### LINKED ISSUE #450: Pydantic Documentation Ideas
After #431 was merged, I have been trying different ways to maintain documentation of the core classes. We have an auto documentation module that provides automatic documentation for all our classes when used in an interpreter shell or an IDE. 

image

Current style:
image

Other alternative is a tabular representation of the Model schema as shown below

image

### PR DESCRIPTION
**Pull request by umesh-timalsina**
Closes #450.

Docs available at https://gmso.mosdef.org/en/450-documentation/data_structures.html (Note: might not work after this PR is merged and this branch is deleted)

### CODE REVIEW

**Review by daico007 — APPROVED** [review]
LGTM

**Review by uppittu11 — APPROVED** [review]
Looks like everything is there

**Comment by daico007** [comment]
I will merge this once all the tests pass

File: gmso/abc/abstract_potential.py [hunk 1]
```diff
@@ -60,10 +60,12 @@ def name(self):
 
     @property
     def independent_variables(self):
+        """Optional[Union[set, str]]\n\tThe independent variables in the expression of the potential"""
         return self.potential_expression_.independent_variables
 
     @property
     def expression(self):
+        """Optional[Union[str, sympy.Expr]]\n\tThe mathematical expression corresponding to the potential"""
```
Comment (justinGilmer) [inline gmso/abc/abstract_potential.py:68]:
```suggestion
        """Optional[Union[str, sympy.Expr]]\n\tThe mathematical expression of the functional form of the potential"""
```

File: gmso/core/parametric_potential.py [hunk 2]
```diff
@@ -70,6 +70,7 @@ def __init__(self,
 
     @property
     def parameters(self):
+        """Optional[dict]\n\tThe parameters of the potential and their values, as unyt quantities"""
```
Comment (justinGilmer) [inline gmso/core/parametric_potential.py:73]:
```suggestion
        """Optional[dict]\n\tThe parameters of the `Potential` expression and their corresponding values, as `unyt` quantities"""
```

File: gmso/abc/abstract_potential.py [hunk 3]
```diff
@@ -60,10 +60,12 @@ def name(self):
 
     @property
     def independent_variables(self):
+        """Optional[Union[set, str]]\n\tThe independent variables in the expression of the potential"""
```
Comment (justinGilmer) [inline gmso/abc/abstract_potential.py:63]:
```suggestion
        """Optional[Union[set, str]]\n\tThe independent variables in the `Potential`'s expression"""
```

**Review by justinGilmer — COMMENTED** [review]
just a few small edits to the docstrings. Feel free to accept/reject or modify. I think these might make it a bit clearer

**Review by justinGilmer — APPROVED** [review]
resolved, LGTM

**Pull request merged**